docs/examples/go-ipfs-as-a-library/main.go

```diff
@@ -0,0 +1,213 @@
+package main
+
+import (
+	"context"
+	"fmt"
+	"io"
+	"io/ioutil"
+	"os"
+	"path/filepath"
+
+	config "github.com/ipfs/go-ipfs-config"
+	files "github.com/ipfs/go-ipfs-files"
+	"github.com/ipfs/go-ipfs/core"
+	"github.com/ipfs/go-ipfs/core/coreapi"
+	"github.com/ipfs/go-ipfs/plugin/loader"
+	"github.com/ipfs/go-ipfs/repo/fsrepo"
+	iCore "github.com/ipfs/interface-go-ipfs-core"
+	iCorePath "github.com/ipfs/interface-go-ipfs-core/path"
+)
+
+type cfgOpt func(*config.Config)
+
+func setupPlugins(path string) error {
+	// Load plugins. This will skip the repo if not available.
+	plugins, err := loader.NewPluginLoader(filepath.Join(path, "plugins"))
+	if err != nil {
+		return fmt.Errorf("error loading plugins: %s", err)
+	}
+
+	if err := plugins.Initialize(); err != nil {
+		return fmt.Errorf("error initializing plugins: %s", err)
+	}
+
+	if err := plugins.Inject(); err != nil {
+		return fmt.Errorf("error initializing plugins: %s", err)
+	}
+
+	return nil
+}
+
+func createTempRepo(ctx context.Context) (string, error) {
+	repoPath, err := ioutil.TempDir("", "ipfs-shell")
+	if err != nil {
+		return "", fmt.Errorf("failed to get temp dir: %s", err)
+	}
+
+	// Set default config with option for 2048 bit key
+	cfg, err := config.Init(ioutil.Discard, 2048)
+	if err != nil {
+		return "", err
+	}
+
+	// configure the temporary node
+	// cfg.Routing.Type = "dhtclient"
+	// cfg.Experimental.QUIC = true
+	cfg.Datastore.Spec = map[string]interface{}{
+		"type": "mem",
```

Comment:
changing to `mem` makes it pass. I learned about it at the default datastore options https://github.com/ipfs/go-ipfs/blob/master/repo/fsrepo/datastores.go#L58-L65

Reply:
What is the trick to get it to support `flatfs` ?

Reply:
The docs https://github.com/ipfs/go-ipfs/blob/master/docs/datastores.md#flatfs only talking about changing the config value as done here.

Reply:
Suggestion to check from @dirkmc (Thank you!): Build & c&p the plugins -- https://github.com/ipfs/go-ipfs/blob/master/docs/plugins.md#in-tree -- into the tmp repo path.

Reply:
We shouldn't need any of that. FlatFS is a preloaded plugin, also non-preloaded plugins don't work on Windows so please don't use that as an example.

Will take a look, but my commit managed to get both FlatFS and Badger to work correctly. I'll see if any changes were made.

Reply:
@Stebalien I was already importing `github.com/ipfs/go-ipfs/plugin/loader` from the beginning and yet, I've always seen that error https://github.com/ipfs/go-ipfs/pull/6695#discussion_r333062224

> Will take a look, but my commit managed to get both FlatFS and Badger to work correctly. I'll see if any changes were made.

@aschmahmann it did not. You were relying on a node spawning on the default repo path which already had all the plugins installed correctly.

Reply:
Update: We got flatfs to load correctly with https://github.com/ipfs/go-ipfs/pull/6695/commits/de61e9a6b45c9faa1416cdc259fe12a549bf3c3a